reference/src/client/test/sa_key_provision.cpp

```diff
@@ -0,0 +1,568 @@
+        if(UNSUPPORTED_KEY == *hmac_key) {
+           GTEST_SKIP() << "key type, size or curve are not supported";
+        }
+        auto exported_hmac_key = export_key(mixin, *hmac_key);
+        ASSERT_NE(nullptr, exported_hmac_key);
+        INFO("export_hmac_key length : %d", exported_hmac_key->size());
+
+        auto sa_hmac_key = create_uninitialized_sa_key();
+        ASSERT_NE(sa_hmac_key,nullptr);
+
+        sa_status status_hmac_key = sa_key_import(sa_hmac_key.get(),
+           SA_KEY_FORMAT_EXPORTED, exported_hmac_key->data(), exported_hmac_key->size(),nullptr);
+        ASSERT_EQ(status_hmac_key, SA_STATUS_OK);
+
+        auto header_hmac_key = key_header(*sa_hmac_key);
+        ASSERT_NE(nullptr, header_hmac_key.get());
+        ASSERT_TRUE(memcmp(&rights, &header_hmac_key->rights, sizeof(sa_rights)) == 0);
+        ASSERT_EQ(clear_hmac_key.size(), header_hmac_key->size);
+        ASSERT_EQ(SA_KEY_TYPE_SYMMETRIC, header_hmac_key->type);
+
+        sa_kdf_parameters_netflix kdf_parameters_netflix = {
+           .kenc = *encryption_key,
+           .khmac = *hmac_key};
+
+        auto key = create_uninitialized_sa_key();
+        ASSERT_NE(nullptr, key);
+        sa_status status = sa_key_derive(key.get(), &rights, SA_KDF_ALGORITHM_NETFLIX, &kdf_parameters_netflix);
+        ASSERT_EQ(SA_STATUS_OK, status);
+        std::vector<uint8_t> clear_wrapping_key(SYM_128_KEY_SIZE);
+        ASSERT_TRUE(netflix_wrapping_key_kdf(clear_wrapping_key, clear_encryption_key, clear_hmac_key));
+        ASSERT_TRUE(key_check_sym(*key, clear_wrapping_key));
+
+        //check wrapping key validation.
+        sa_rights_set_allow_all(&rights);
+        auto wrapping_key = create_sa_key_symmetric(&rights, clear_wrapping_key);
+        ASSERT_NE(nullptr, wrapping_key);
+        if(UNSUPPORTED_KEY == *wrapping_key) {
+           GTEST_SKIP() << "key type, size or curve are not supported";
+        }
+        auto exported_wrapping_key = export_key(mixin, *wrapping_key);
+        ASSERT_NE(nullptr, exported_wrapping_key);
+        INFO("export_wrapping_key length : %d", exported_wrapping_key->size());
+
+        auto sa_wrapping_key = create_uninitialized_sa_key();
+        ASSERT_NE(sa_wrapping_key,nullptr);
+
+        sa_status status_wrapping_key = sa_key_import(sa_wrapping_key.get(), 
+           SA_KEY_FORMAT_EXPORTED, exported_wrapping_key->data(), exported_wrapping_key->size(),nullptr);
+        ASSERT_EQ(status_wrapping_key, SA_STATUS_OK);
+
+        auto header_wrapping_key = key_header(*sa_wrapping_key);
+        ASSERT_NE(nullptr, header_wrapping_key.get());
+        ASSERT_TRUE(memcmp(&rights, &header_wrapping_key->rights, sizeof(sa_rights)) == 0);
+        ASSERT_EQ(clear_wrapping_key.size(), header_wrapping_key->size);
+        ASSERT_EQ(SA_KEY_TYPE_SYMMETRIC, header_wrapping_key->type);
+    }
+#endif //KEY_VALIDATION_TEST
+
+#ifdef FILE_BASED_FETCH_KEY
+ /*there are two file 0351000003510001.key and 0651000006510001.bin under
```

Comment:
Let's not leak Comcast specific implementations through open source implementations.